packages/tizen_notification/tizen/src/tizen_notification_plugin.cc

```diff
@@ -0,0 +1,525 @@
+                  std::vector<const char *> values;
+                  std::vector<std::string> dummy;
+                  std::string value;
+                  if (GetValueFromArgs(&extra_value, map_key.c_str(),
+                                       value_list)) {
+                    for (size_t i = 0; i < value_list.size(); i++) {
+                      dummy.push_back(std::get<std::string>(value_list[i]));
+                    }
+                    for (size_t i = 0; i < value_list.size(); i++) {
+                      values.push_back(dummy[i].c_str());
+                    }
+                    app_control_add_extra_data_array(
+                        app_control, map_key.c_str(), values.data(),
+                        values.size());
+                  } else if (GetValueFromArgs(&extra_value, map_key.c_str(),
+                                              value)) {
+                    app_control_add_extra_data(app_control, map_key.c_str(),
+                                               value.c_str());
+                  }
+                }
+              }
+              ret = notification_set_launch_option(
+                  noti_handle, NOTIFICATION_LAUNCH_OPTION_APP_CONTROL,
+                  (void *)app_control);
+              if (ret != NOTIFICATION_ERROR_NONE) {
+                FreeNotification(noti_handle);
+                DestroyAppControl(app_control);
+                LOG_ERROR("notification_set_launch_option failed : %s",
+                          get_error_message(ret));
+                result->Error("notification_set_launch_option failed",
+                              std::string(get_error_message(ret)));
+                return;
+              }
+            }
+          }
+        }
+
+        if (app_control) {
+          ret = app_control_destroy(app_control);
+          if (ret != APP_CONTROL_ERROR_NONE) {
+            FreeNotification(noti_handle);
+            LOG_ERROR("app_control_destroy failed : %s",
+                      get_error_message(ret));
+            result->Error("app_control_destroy failed",
+                          std::string(get_error_message(ret)));
+            return;
+          }
+        }
+        ret = notification_post(noti_handle);
+        if (ret != NOTIFICATION_ERROR_NONE) {
+          FreeNotification(noti_handle);
+          LOG_ERROR("notification_post failed : %s", get_error_message(ret));
+          result->Error("notification_post failed",
+                        std::string(get_error_message(ret)));
+          return;
+        }
+        ret = notification_free(noti_handle);
+        if (ret != NOTIFICATION_ERROR_NONE) {
+          LOG_ERROR("notification_free failed : %s", get_error_message(ret));
+          result->Error("notification_free failed",
+                        std::string(get_error_message(ret)));
+          return;
+        }
+        result->Success();
+      }
+    } else if (method_call.method_name().compare("cancel") == 0) {
```

Comment:
```suggestion
      }
      result->Success();
    } else if (method_call.method_name().compare("cancel") == 0) {
```

The method handler should always send a response.

If you want to be safe, you can send an error response when the arguments are invalid.